mapper/tedge_json/src/lib.rs

```diff
@@ -0,0 +1,287 @@
+                    }
+                }
+                if measurements.is_empty() {
+                    Err(ThinEdgeJsonError::EmptyThinEdgeJsonRoot)
+                } else {
+                    Ok(ThinEdgeJson {
+                        timestamp: time,
+                        values: measurements,
+                    })
+                }
+            }
+            _ => Err(ThinEdgeJsonError::new_invalid_json_root(&input)),
+        }
+    }
+
+    fn check_timestamp_for_iso8601_complaint(
+        value: &JsonValue,
+    ) -> Result<String, ThinEdgeJsonError> {
+        match value {
+            JsonValue::Short(str) => {
+                let timestamp = str.as_str();
+                //Parse fails if timestamp is not is8601 complaint
+                DateTime::parse_from_rfc3339(&timestamp).map_err(|err| {
+                    ThinEdgeJsonError::InvalidTimestamp {
+                        value: String::from(timestamp),
+                        from: err,
+                    }
+                })?;
+                Ok(String::from(timestamp))
+            }
+            _ => Err(ThinEdgeJsonError::new_invalid_json_time(value)),
+        }
+    }
+}
+
+impl SingleValueMeasurement {
+    pub fn new(name: &str, value: f64) -> Result<Self, ThinEdgeJsonError> {
+        if value == 0.0 || value.is_normal() {
+            let single_value = SingleValueMeasurement {
+                name: String::from(name),
+                value,
+            };
+            Ok(single_value)
+        } else {
+            Err(ThinEdgeJsonError::InvalidThinEdgeJsonNumber {
+                name: String::from(name),
+            })
+        }
+    }
+}
+
+impl MultiValueMeasurement {
+    pub fn new(
+        name: &str,
+        multi_value_obj: &json::object::Object,
+    ) -> Result<Self, ThinEdgeJsonError> {
+        let mut single_values = vec![];
+
+        for (k, v) in multi_value_obj.iter() {
+            match v {
+                JsonValue::Number(num) => {
+                    //Single Value object
+                    let single_value_measurement = SingleValueMeasurement::new(k, (*num).into())?;
+                    single_values.push(single_value_measurement);
+                }
+                JsonValue::Object(_object) => {
+                    return Err(ThinEdgeJsonError::InvalidThinEdgeHierarchy {
+                        name: String::from(k),
+                    })
+                }
+                value => {
+                    return Err(ThinEdgeJsonError::new_invalid_json_value(name, value));
+                }
+            }
+        }
+        if single_values.is_empty() {
+            Err(ThinEdgeJsonError::EmptyThinEdgeJson {
+                name: String::from(name),
+            })
+        } else {
+            Ok(MultiValueMeasurement {
+                name: String::from(name),
+                values: single_values,
+            })
+        }
+    }
+}
+
+pub fn input_prefix(input: &str, len: usize) -> String {
```

Comment:
This is an helper function and should not be `pub`.

=> The tests have to be moved from `c8y_translator_lib/lib.rs` to here, not that function made `pub` to please Cargo.

Reply:
fixed